### PR TITLE
Remove idCounter and allow BSS deals & facilities to use ObjectIDs

### PR DESCRIPTION
**Pull request by IgnatG**
Currently, BSS/EWCS deals and facilities generate a special _id format, e.g: **1000023** Each time a new deal or facility is created, we check the latest _id and increment from that.

This was put in place at the very beginning of the project because it was the required _id format for Workflow integration. We don't have Workflow integration anymore.

This PR removes this special _id formatting/generation and aligns _id generation with standard MongoDB _id generation.

It also means that GEF and BSS deals and facility _ids are now aligned 🎉 

So instead of this:
- `1000023`

It's now using ObjectIds:
- `ObjectId("6203aa9c1322fc0012a228f1")`

### CODE REVIEW

File: e2e-tests/portal/cypress/integration/journeys/number-generator/test-data/baseBssDeal.js [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  details: {},
+  submissionDetails: {
+    'indemnifier-address-country': {
+      code: 'GBR',
+      name: 'United Kingdom',
+    },
+    'indemnifier-address-line-3': '',
+    'indemnifier-address-line-1': '27a',
+    'indemnifier-address-line-2': 'Maxwell Road',
+    'indemnifier-address-postcode': 'HA6 2XY',
```
Comment (abhi-markan) [inline e2e-tests/portal/cypress/integration/journeys/number-generator/test-data/baseBssDeal.js:37]: Anonymise the post code please.

File: e2e-tests/portal/cypress/integration/journeys/number-generator/test-data/baseBssDeal.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    'indemnifier-correspondence-address-country': {
+      code: 'GBR',
+      name: 'United Kingdom',
+    },
+    'indemnifier-correspondence-address-line-3': 'Essex',
+    'indemnifier-correspondence-address-line-1': '27 Petersfield',
+    'indemnifier-correspondence-address-line-2': '',
+    'indemnifier-correspondence-address-postcode': 'CM1 4EP',
+    'indemnifier-correspondence-address-town': 'Chelmsford',
+    'indemnifier-name': 'WATKINSON TRADING LIMITED',
```
Comment (abhi-markan) [inline e2e-tests/portal/cypress/integration/journeys/number-generator/test-data/baseBssDeal.js:49]: Change to `TEST TRADING LIMITED`

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Please add description.

File: e2e-tests/portal/cypress/integration/journeys/number-generator/test-data/baseBssDeal.js (hunk 1 above)
Comment (ttbarnes) [reply e2e-tests/portal/cypress/integration/journeys/number-generator/test-data/baseBssDeal.js]: Done - there's some other files that don't have anonymised postcode/towns - can update these in another PR.

**Review by abhi-markan — APPROVED** [review]
Great work :-)

**Pull request merged**